scannerlib/configchecks/sql_check.go

```diff
@@ -61,27 +61,31 @@ func (c *MySQLCheck) Exec() (ComplianceMap, error) {
 }
 
 // BenchmarkIDs returns the IDs of the benchmarks associated with this check.
-func (c *MySQLCheck) BenchmarkIDs() []string {
-	// We don't do batching for MySQL checks, so we will always have exactly one ID.
+func (c *SQLCheck) BenchmarkIDs() []string {
+	// We don't do batching for SQL checks, so we will always have exactly one ID.
 	return []string{c.benchmarkID}
 }
 
-func (c *MySQLCheck) String() string {
-	return fmt.Sprintf("[MySQL check with id %q]", c.benchmarkID)
+func (c *SQLCheck) String() string {
+	return fmt.Sprintf("[SQL check with id %q]", c.benchmarkID)
 }
 
 // createSQLChecksFromConfig parses the benchmark config and creates the executable
 // SQL checks that it defines.
-func createSQLChecksFromConfig(ctx context.Context, benchmarks []*benchmark, timeout *timeoutOptions, sq scanapi.SQLQuerier) ([]*MySQLCheck, error) {
+func createSQLChecksFromConfig(ctx context.Context, benchmarks []*benchmark, timeout *timeoutOptions, sq scanapi.SQLQuerier) ([]*SQLCheck, error) {
 	// TODO(b/235991635): Use timeout.
-	checks := []*MySQLCheck{}
+	checks := []*SQLCheck{}
 	for _, b := range benchmarks {
 		for _, alt := range b.alts {
 			for _, sqlCheckInstruction := range alt.proto.GetSqlChecks() {
-				if sqlCheckInstruction.GetTargetDatabase() != ipb.SQLCheck_DB_MYSQL {
-					return nil, errors.New("only MySQL/MariaDB database checks are supported")
+				dbtype, err := sq.SupportedDatabase()
+				if err != nil {
+					return nil, err
 				}
-				checks = append(checks, &MySQLCheck{
+				if dbtype != sqlCheckInstruction.GetTargetDatabase() {
```

Comment:
Hm, I wonder if we should do this check inside the SQLQuery() implementation instead. The downside is that we'll return the error at check time instead of config creation time, but the upside is that we don't have to add SupportedDatabase() to the API (it can be a private function).

Reply:
We thought that the proposed implementation of adding `SupportedDatabase()` would be the least impactful one, since your proposal would require modifying the signature of the existing `SQLQuery()` function in order to add the database type as an argument.